src/main/java/blusunrize/immersiveengineering/common/blocks/metal/FluidPipeTileEntity.java

```diff
@@ -579,19 +624,24 @@ public boolean receiveClientEvent(int id, int arg)
 	@Override
 	public VoxelShape getCollisionShape(ISelectionContext ctx)
 	{
-		return SHAPES.get(new BoundingBoxKey(true, this));
+		return SHAPES.get(new BoundingBoxKey(false, this));
 	}
 
 	@Override
 	public VoxelShape getSelectionShape(@Nullable ISelectionContext ctx)
 	{
-		return SHAPES.get(new BoundingBoxKey(false, this));
+		boolean hasHammer;
+		if(ctx!=null&&ctx.hasItem(Tools.hammer))
+			hasHammer = true;
+		else
+			hasHammer = false;
```

Comment:
I would prefer this to use `Utils.isHammer(ItemStack)`, however you will need to check the ctx to be EntitySelectionContext and then retrieve the MAIN_HAND item of the entity involved.

That is more complex, but our utility method should be used whereever possible, so that if our logic for detecting a hammer changes (currently using tool classes, may allow tags in the future), we only have to change it in one place.

Reply:
neat. Done!